### PR TITLE
fix: trigger onError callback when loading google pay js failed

### PR DESCRIPTION
**Pull request by shirly-chen-awx**
Background: https://pay.google.com/gp/p/js/pay.js will be failed to load in china if the user don't have the agent software. It's better to call the onError callback when loading js errored.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,6 +25,6 @@ yarn-error.log*
 
 *.orig
 .env
-
+.history
```
Comment (JlUgia) [inline .gitignore:28]: Suggestion to remove this folder if it's specific to your setup.

File: src/lib/button-manager.ts [hunk 2]
```diff
@@ -76,9 +76,18 @@ export class ButtonManager {
   }
 
   async mount(element: Element): Promise<void> {
-    if (!this.isGooglePayLoaded()) {
-      await loadScript('https://pay.google.com/gp/p/js/pay.js');
+    try {
```
Comment (JlUgia) [inline src/lib/button-manager.ts:79]: Suggestion to keep the if clause outside of the try block.

**Comment by JlUgia** [comment]
Hi @shirly-chen-awx, thank you for your PR.
I added two comments for your review.

**Comment by shirly-chen-awx** [comment]
Hi @JlUgia ,  thanks for your review, the pr updated

File: src/lib/button-manager.ts [hunk 3]
```diff
@@ -77,9 +77,19 @@ export class ButtonManager {
...
+      } catch(err) {
+        if (this.config?.onError) {
+          this.config.onError(err as Error);
+        } else {
+          console.error(err);
+        }
+      }
     }
 
+
```
Comment (JlUgia) [inline src/lib/button-manager.ts:91]: Let's also bring this back to one space separation by removing the two new lines added.

**Comment by JlUgia** [comment]
Thank you for the update @shirly-chen-awx.
Two final comments and we'll be good to go.

**Review by socsieng — APPROVED** [review]

**Comment by shirly-chen-awx** [comment]
@JlUgia the build failed, can you help check with that?

**Comment by shirly-chen-awx** [comment]
Hi, @socsieng 
can you help check with failed build? It seems it was broken several months ago.

**Comment by socsieng** [comment]
> Hi, @socsieng
> can you help check with failed build? It seems it was broken several months ago.

Looks like it's failing here: https://github.com/google-pay/google-pay-button/runs/6650864917?check_suite_focus=true#step:7:22

> ```
> Checking formatting...
> [warn] examples/react/src/serviceWorker.ts
> [warn] Code style issues found in the above file(s). Forgot to run Prettier?
> ```

Working locally for me, but might be worth running `npm run pretty` to reformat all files.

**Comment by socsieng** [comment]
Created https://github.com/google-pay/google-pay-button/pull/143 to fix the prettier issue.

**Pull request merged**